python-spec/src/somacore/data.py

```diff
@@ -185,7 +190,7 @@ def create(
         uri: str,
         *,
         type: pa.DataType,
-        shape: Sequence[int],
+        shape: Sequence[Union[int, None]],
```

Comment:
Is this what we want instead?
```suggestion
        shape: Union[Sequence[int], None],
```

Reply:
@atolopko-czi  I also found a simple `None` more appealing, but, no because we need _somewhere_ to get the "N" in "N-D" :(

Reply:
`Sequence[Optional[int]]` is more idiomatic and represents the "it can be a value or None" meaning.

Reply:
So we allow `shape = [None]`? Why? 

Or is this because we want to allow for _some_ of the dimensions to be optional? `shape = [None, 3]`

Reply:
@atolopko-czi  if people want a 2-D array or a 3-D array we have no place in the API for them to put the number 2 or 3. All we have is `(None, None)` or `(None, None, None)`.

Reply:
@atolopko-czi I can add a comment to make that clearer b/c surely other folks will wonder the same thing